client/src/main/kotlin/io/spine/examples/chatspn/chat/ChatsPage.kt

```diff
@@ -0,0 +1,172 @@
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import io.spine.examples.chatspn.ChatColors
+import io.spine.examples.chatspn.ChatId
+import io.spine.examples.chatspn.ChatProvider
+import io.spine.examples.chatspn.UserProvider
+
+/**
+ * Represents the 'Chats' page in the application.
+ */
+@Composable
+@Preview
+fun ChatsPage(userProvider: UserProvider, chatProvider: ChatProvider) {
+    val chats by chatProvider.chats().collectAsState()
+    var selectedChat by remember { mutableStateOf(ChatId.getDefaultInstance()) }
+    Row {
+        Surface(
+            elevation = 8.dp
+        ) {
+            Column(
+                Modifier
+                    .fillMaxHeight()
+                    .width(280.dp)
+            ) {
+                UserProfilePanel(userProvider.loggedUser())
+                UserSearchBar()
+                LazyColumn( // Chats list
+                    modifier = Modifier.fillMaxSize(),
+                    verticalArrangement = Arrangement.spacedBy(8.dp)
+                ) {
+                    chats.forEachIndexed { index, chat ->
+                        item(key = index) {
+                            ChatPreviewPanel(
+                                chatName(chat, userProvider),
+                                chat.lastMessage.content,
+                                chat.id.equals(selectedChat)
+                            ) {
+                                selectedChat = chat.id
+                            }
+                        }
+                    }
+                    item {
+                        Box(Modifier.height(50.dp))
+                    }
+                }
+            }
+        }
+        val isChatSelected = chats
+            .stream()
+            .map { chat -> chat.id }
+            .anyMatch { id -> id.equals(selectedChat) }
+        if (!isChatSelected) {
+            Box(
+                modifier = Modifier.fillMaxSize(),
+                contentAlignment = Alignment.Center
+            ) {
+                Text(
+                    text = "Choose the chat",
+                    color = ChatColors.SECONDARY,
+                    fontSize = 20.sp
+                )
+            }
+        } else {
```

Comment:
Same goes for the long blocks/functions. We don't want them this long. Aim for extracting modular pieces, naming them, and documenting instead.